ee/katc/config.go

```diff
@@ -0,0 +1,120 @@
+package katc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"runtime"
+
+	"github.com/osquery/osquery-go"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+// katcSourceType defines a source of data for a KATC table. The `name` is the
+// identifier parsed from the JSON KATC config, and the `dataFunc` is the function
+// that performs the query against the source.
+type katcSourceType struct {
+	name     string
+	dataFunc func(ctx context.Context, slogger *slog.Logger, path string, query string, sourceConstraints *table.ConstraintList) ([]sourceData, error)
+}
+
+// sourceData holds the result of calling `katcSourceType.dataFunc`. It maps the
+// source to the query results. (A config may have wildcards in the source,
+// allowing for querying against multiple sources.)
+type sourceData struct {
+	path string
+	rows []map[string][]byte
+}
+
+const (
+	sqliteSourceType = "sqlite"
+)
+
+func (kst *katcSourceType) UnmarshalJSON(data []byte) error {
+	var s string
+	err := json.Unmarshal(data, &s)
```

Comment:
What is the purpose of doing a json.Unmarshall here? If I understand correctly, were just passing a single word string as []bytes? Is it just to remove the quotes? That's okay I suppose, could you please leave a comment?

Reply:
Yes, we are just processing a single-word string here. This means we use the official implementation for unmarshalling JSON strings, which felt a little more correct than casting to bytes. It's also how I've seen it done in examples and in the docs -- e.g. see the `Example (CustomMarshalJSON)` at the top [here](https://pkg.go.dev/encoding/json#pkg-overview).